linux_os/guide/system/software/sudo/sudo_require_reauthentication/oval/shared.xml

```diff
@@ -0,0 +1,25 @@
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("'Ensure sudo timestamp_timeout is appropriate - sudo timestamp_timeout") }}}
+    <criteria comment="The timestamp_timeout should be configured" >
+      <criterion comment="check configuration in /etc/sudoers" test_ref="test_sudo_timestamp_timeout" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="check correct configuration in /etc/sudoers" id="test_sudo_timestamp_timeout" version="1">
+    <ind:object object_ref="obj_sudo_timestamp_timeout"/>
+    <ind:state state_ref="state_sudo_timestamp_timeout" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="obj_sudo_timestamp_timeout" version="1">
+    <ind:filepath>/etc/sudoers</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*Defaults[\s]+timestamp_timeout=([\d]+)$</ind:pattern>
+    <ind:instance datatype="int" operation="greater than or equal">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_state id="state_sudo_timestamp_timeout"
+  version="1">
+    <ind:subexpression datatype="int" operation="greater than or equal">0</ind:subexpression>
```

Comment:
You need to use the variable you have just created here. See this: https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_pam_unix_rounds_system_auth/oval/shared.xml#L31

Reply:
Actually it doesn't make sense to use the variable here in the OVAL since any positive value should be valid.

Reply:
Yes. The OVAL code is not needed here because ansible and OVAL code are genererated by template. I will do the fix.

Reply:
Commited fix.

Reply:
In my opinion we need to have this custom OVAL here. The OVAL generated by the template has a different subexpression comparison and that will not work with this rule (this rule is about checking positive integers). That's why I suggested the `oval: "off"` in the template data section.

See this: https://github.com/ComplianceAsCode/content/blob/master/shared/templates/sudo_defaults_option/oval.template#L22

You can revert back to the original implementation.

Reply:
Reverted previous commit and did recommended fix.

Reply:
Reverted OVAL check for variable.